src/libasr/pass/pass_list_concat.cpp

```diff
@@ -0,0 +1,114 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/pass_list_concat.h>
+#include <libasr/pass/pass_utils.h>
+
+#include <vector>
+#include <utility>
+
+
+namespace LFortran {
+
+using ASR::down_cast;
+using ASR::is_a;
+
+/*
+This ASR pass replaces list-concat to
+
+Converts:
+    x = i + j // lists
+
+to:
+
+ */
+
+
+class ReplaceListConcat : public ASR::BaseExprReplacer<ReplaceListConcat>
+{
+    private:
+
+        Allocator& al;
+        ASR::TranslationUnit_t &unit;
+
+    public:
+        ReplaceListConcat(Allocator &al_, ASR::TranslationUnit_t &unit_) :
+        al(al_), unit(unit_)
+        { }
+
+        void create_concat_function() {
+            SymbolTable *fn_scope = al.make_new<SymbolTable>(unit.m_global_scope);
+            std::string fn_name = "_lcompilers_list_concat";
+            ASR::ttype_t *type;
+            Location loc;
+            ASR::asr_t *return_var=nullptr;
+            ASR::expr_t *return_var_ref=nullptr;
+            char *var_name;
+            int idx = 1;
+
+            ASR::asr_t *fn = ASR::make_Function_t(
+            al, loc,
+            /* a_symtab */ fn_scope,
+            /* a_name */ (char *)fn_name.c_str(),
+            /* a_args */ nullptr,
+            /* n_args */ 0,
+            /* a_body */ nullptr,
+            /* n_body */ 0,
+            /* a_return_var */ return_var_ref,
+            ASR::abiType::BindC,
+            ASR::Public, ASR::Implementation,
+            nullptr,
+            false, false, false, false, false,
+            nullptr, 0,
+            nullptr, 0,
+            false);
+            unit.m_global_scope->add_symbol(fn_name, down_cast<ASR::symbol_t>(fn));
+        }
+
+        void replace_ListConcat(const ASR::ListConcat_t* x) {
+            Location loc = x->base.base.loc;
+            Vec<ASR::call_arg_t> a_args;
+            a_args.reserve(al, 2);
+            ASR::call_arg_t left_list, right_list;
+            left_list.loc = x->m_left->base.loc;
+            left_list.m_value = x->m_left;
+            a_args.push_back(al, left_list);
+            right_list.loc = x->m_right->base.loc;
+            right_list.m_value = x->m_right;
+            a_args.push_back(al, right_list);
+            create_concat_function();
+            *current_expr = ASRUtils::EXPR(ASR::make_FunctionCall_t(al, loc,
+                                           nullptr, nullptr, a_args.p, 2,
```

Comment:
Later, use that function `_lcompilers_list_concat` here to concat the list, right?